### PR TITLE
Navbar, Avatar, Menu, Switch

### LINKED ISSUE #26: Implement Navbar
- Title
- Logo
- Version
- Left content
- Right content
- User dropdown (separate component?)
- Mobile support / breakpoints / resizeobserver callback?

### PR DESCRIPTION
**Pull request by AaronPlave**
Adds several new components:

- Navbar (Custom implementation, responsive and composable)
- Avatar (Radix)
- Menu (Dropdown Menu in Radix)
- Switch (Radix)

Closes #26 

TODO:
- [x] Brand link
- [x] Try reworking navbar to use a `Navbar.Breakpoint` approach instead of bundling breakpoints into `Navbar.Content`. 
- [x] Remove large Avatar
- [x] Fix Checkbox and Switch label positioning
- [x] Add docs and list examples for Checkbox and Switch to illustrate designed usage wrt parent width

### CODE REVIEW

**Comment by anilnatha** [comment]
Hey @AaronPlave, thanks for taking the time to add these components, these look great!

I spent a little time reviewing them in storybook and was wondering if the following items require an update:

### Switch

1. The `labelPosition` values are rendering the opposite of what is chosen. In other words, when one chooses `left`, the label is on the right.

Screen Shot 2023-05-29 at 19 16 22

2. I see that the component's width is inherited the parent container; however, a flex width spacing is used between the label and the switch when `labelPosition="right"`; however, when the `labelPosition="left"`, the label and switch are placed next to each other. My thought is that they should always be next to one another. Also, should their width be inherited from their parent container or should the switch's contents dictate their width?

Screen Shot 2023-05-29 at 19 16 50

### Avatar

1. Perhaps the typography sizing should scale up when the `large` version is used? At the moment both avatar sizes use the same type size.

**Comment by AaronPlave** [comment]
**Switch**
1. Good catch on label position - I'll fix that for Switch and Checkbox. 
2. For the situation where text is left, toggle is right, the Stellar figma design specifies that the width be defined by the parent element. https://www.figma.com/file/a696svN2S7YNlZRYAkeLob/Stellar-Design-System?type=design&node-id=5441-7943&t=tLux7kDqdl3Ig71o-4. I think the idea behind this is to maintain a consistent positioning of the right-side toggle in a list of switches/checkboxes/radios. The stories for Switch and Checkbox present their components in a sort of unusual fashion, maybe it'd be useful to constrain the parent width in the stories to show intended use? 

**Avatar**
1. Good point. I also thought about the text sizing and could go either way on it since we don't have a design spec for it. One argument against larger text size here would be that large buttons in Stellar don't have larger text size. Guess it depends on where this large avatar will be used. Could alternatively just get rid of the large size until we have a spec? Wanted to introduce it just for the user dropdown at first which only uses the medium/default size.

**Comment by anilnatha** [comment]
Hey @AaronPlave , regarding your comments:

> maybe it'd be useful to constrain the parent width in the stories to show intended use?

I think this would be helpful.

>One argument against larger text size here would be that large buttons in Stellar don't have larger text size. Guess it depends on where this large avatar will be used. Could alternatively just get rid of the large size until we have a spec?

Hmm, is there a design requirement to not scale the font on the larger buttons? I could imagine that the typography should scale when buttons do.  I'm good with hiding the large size until we have a spec or consensus on how components across the design system should behave when their size is increased or decreased.

**Comment by rtapella** [comment]
Do we need a "logged out" navbar? eg. with a `Log in` button in place of the avatar menu?

**Comment by AaronPlave** [comment]
> Do we need a "logged out" navbar? eg. with a `Log in` button in place of the avatar menu?
> 
> 

Some applications may but don't think stellar necessarily needs to ship that. Seems fairly application specific and the navbar is designed to let you put whatever content you want inside of it. The navbar in the story isn't a single component but instead a collection of stellar and components plus some custom content from the story, similar to how an application would compose a navbar for specific use.

**Comment by rtapella** [comment]
Also... in mobile mode, does it make sense to show the account/login button + the hamburger?

File: src/components/Avatar/Avatar.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import classNames from "classnames";
+import * as AvatarPrimitive from "@radix-ui/react-avatar";
+
+export type AvatarProps = {
+  src?: string;
+  text?: string;
+  size?: "medium" | "large";
+  className?: string;
+} & AlertDialog.DialogProps;
```
Comment (duranb) [inline src/components/Avatar/Avatar.tsx]: Does `AlertDialog` need to be imported? If not, I'm probably just not seeing what is putting the namespace on the global scope.

**Review by duranb — CHANGES_REQUESTED** [review]

File: src/components/Avatar/Avatar.tsx (hunk 1 above)
Comment (AaronPlave) [reply src/components/Avatar/Avatar.tsx]: Whoops! Good catch, came from duplicated component, will fix.

**Comment by AaronPlave** [comment]
> Also... in mobile mode, does it make sense to show the account/login button + the hamburger?

It might? Happy to include it in the example though wasn't sure if it might have a different design?

**Comment by anilnatha** [comment]
@AaronPlave I pulled down your recent changes and reviewed them. Thanks for fixing the label placement issue for the Switch/Toggle component, looks good!

On the Navbar and Menu components, there appears to be an issue with the foreground color for the typography and icon? They show as light color which is very hard to discern. Apologies if you're already working on a resolution for this.

**Comment by duranb** [comment]
> On the Navbar and Menu components, there appears to be an issue with the foreground color for the typography and icon? They show as light color which is very hard to discern. Apologies if you're already working on a resolution for this.

@anilnatha I think that might be part of a separate issue that'll need to be addressed in the stellar library itself. The "dark" CSS theme has background/text color specifications that are both light. Can you confirm if you're using the dark mode theme in your storybook preview? If so, do you mind switching to light mode and seeing if it's still too light?

**Comment by anilnatha** [comment]
> > On the Navbar and Menu components, there appears to be an issue with the foreground color for the typography and icon? They show as light color which is very hard to discern. Apologies if you're already working on a resolution for this.
> 
> @anilnatha I think that might be part of a separate issue that'll need to be addressed in the stellar library itself. The "dark" CSS theme has background/text color specifications that are both light. Can you confirm if you're using the dark mode theme in your storybook preview? If so, do you mind switching to light mode and seeing if it's still too light?

@duranb Thanks for the summary of what the issue may be. I was using the dark mode when viewing the changes and verified that the issue is specific to that as evidenced by these screenshots:

### Dark Mode — menu icon not visible

image

### Light Mode — menu icon is now visible

image

**Comment by AaronPlave** [comment]
Dark theming isn't built out for react-stellar yet 😅 but we'll get there in a separate effort. Any of the dark theming you see now is coming from the base Stellar classes

**Comment by anilnatha** [comment]
Right, I remember you mentioning that. So that last issue I reported can be pushed off. I need to be mindful of running this in light mode. My apologies.

**Comment by AaronPlave** [comment]
Any other comments on this PR?

**Comment by duranb** [comment]
> @duranb Thanks for the summary of what the issue may be. I was using the dark mode when viewing the changes and verified that the issue is specific to that as evidenced by these screenshots:

@anilnatha Thank you for verifying!

**Comment by anilnatha** [comment]
> Any other comments on this PR?

None from me.

**Review by anilnatha — APPROVED** [review]

**Review by duranb — APPROVED** [review]

**Pull request merged**